src/components/BottomSheet/BottomSheetMenu.js

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import {StyleSheet, Text, TouchableOpacity} from 'react-native';
+import RBSheet from 'react-native-raw-bottom-sheet';
+import {colors} from '../../utils/colors';
+import dimen from '../../utils/dimen';
+import {normalizeFontSize} from '../../utils/fonts';
+import Gap from '../Gap';
+
+const BottomSheetMenu = ({refSheet, dataSheet = [], height = 234}) => {
+  return (
+    <RBSheet
+      ref={refSheet}
+      closeOnDragDown={true}
+      closeOnPressMask={true}
+      customStyles={{
+        container: styles.rbsheetContainer(height),
+        draggableIcon: styles.rbsheetDraggableIcon
+      }}>
+      {dataSheet.map((item) => (
+        <TouchableOpacity
+          key={item.id}
+          style={item.id === 1 ? styles.rbsheetContent : styles.rbsheetContent2}
+          onPress={item.onPress}>
+          {item.icon}
+          <Gap width={dimen.normalizeDimen(12)} />
+          <Text style={[styles.textShare, item.style]}>{item.name}</Text>
+        </TouchableOpacity>
+      ))}
+    </RBSheet>
+  );
+};
+
+export default BottomSheetMenu;
+
+const styles = StyleSheet.create({
+  rbsheetContainer: (height) => ({
+    backgroundColor: colors.white,
+    borderTopLeftRadius: dimen.normalizeDimen(12),
+    borderTopRightRadius: dimen.normalizeDimen(12),
+    height: dimen.normalizeDimen(height)
+  }),
+  rbsheetDraggableIcon: {
+    backgroundColor: colors.greySubtile1,
+    height: dimen.normalizeDimen(4),
+    width: dimen.normalizeDimen(60),
+    marginTop: dimen.normalizeDimen(16),
+    marginBottom: dimen.normalizeDimen(24)
+  },
+  rbsheetContent: {
+    flexDirection: 'row',
+    alignItems: 'center',
+    borderTopWidth: 1,
+    borderBottomWidth: 1,
+    paddingVertical: dimen.normalizeDimen(14),
+    paddingHorizontal: dimen.normalizeDimen(20),
+    borderColor: colors.greySubtile2
+  },
+  rbsheetContent2: {
+    flexDirection: 'row',
+    alignItems: 'center',
+    borderBottomWidth: 1,
+    paddingVertical: dimen.normalizeDimen(14),
+    paddingHorizontal: dimen.normalizeDimen(20),
+    borderColor: colors.greySubtile2
+  },
+  textShare: {
+    fontSize: normalizeFontSize(16),
+    fontWeight: '400',
+    lineHeight: dimen.normalizeDimen(24),
+    textAlignVertical: 'center'
+  }
+});
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `BottomSheetMenu` component looks good overall. However, there's a potential issue with the way you're handling keys for your mapped elements on line 21. You're using `item.id` as the key, which assumes that every item in `dataSheet` will have a unique id. If two items have the same id or if an item doesn't have an id, React will throw a warning and it could lead to unexpected behavior. 

To fix this, you could use the index from the map function as the key, but only do this if the list is static and won't be reordered or filtered. If the list can change, you should ensure that each item has a unique id.

```diff
      {dataSheet.map((item, index) => (
        <TouchableOpacity
-21:           key={item.id}
+21:           key={index}
          style={item.id === 1 ? styles.rbsheetContent : styles.rbsheetContent2}
          onPress={item.onPress}>
          {item.icon}
          <Gap width={dimen.normalizeDimen(12)} />
          <Text style={[styles.textShare, item.style]}>{item.name}</Text>
        </TouchableOpacity>
      ))}
```

Also, on line 22, you are applying different styles based on whether `item.id` equals 1. This seems like a magic number and could make the code harder to understand for other developers. Consider defining a constant with a descriptive name for this value or adding a comment explaining why you're checking for `item.id === 1`.


<!-- This is an auto-generated comment by OSS CodeRabbit -->